db/migrate/20200313163242_rename_prefix_tables.rb

```diff
@@ -1,24 +1,22 @@
 class RenamePrefixTables < ActiveRecord::Migration[5.2]
   def change
-    safety_assured do
-      rename_column :prefix, :created, :created_at
-      rename_column :prefix, :prefix, :uid
-      remove_column :prefix, :version, :integer
-      remove_column :prefix, :updated, :datetime
-      rename_table :prefix, :prefixes
+    rename_column :prefix, :created, :created_at
+    rename_column :prefix, :prefix, :uid
+    remove_column :prefix, :version, :integer
+    remove_column :prefix, :updated, :datetime
+    rename_table :prefix, :prefixes
 
-      rename_column :allocator_prefixes, :allocator, :provider_id
-      rename_column :allocator_prefixes, :prefixes, :prefix_id
-      add_column :allocator_prefixes, :uid, :string, null: false
-      rename_table :allocator_prefixes, :provider_prefixes
-      add_index :provider_prefixes, [:uid], length: 128
+    rename_column :allocator_prefixes, :allocator, :provider_id
```

Comment:
When running the migration this fails:

```
ActiveRecord::StatementInvalid: Departure::Error: No foreign keys reference `datacite`.`allocator_prefixes`; ignoring --alter-foreign-keys-method.
The new table `datacite`.`_allocator_prefixes_new` does not have a PRIMARY KEY or a unique index which is required for the DELETE trigger.
: ALTER TABLE `allocator_prefixes` DROP PRIMARY KEY

```

I think its because there is no table `allocator_prefixes ` but I am not sure, what do you think?

Reply:
The `allocator_prefixes` table was renamed to `provider_prefixes` a few months ago.